itests/kit/evm.go

```diff
@@ -237,18 +244,25 @@ func (e *EVM) ComputeContractAddress(deployer ethtypes.EthAddress, nonce uint64)
 	return *(*ethtypes.EthAddress)(hasher.Sum(nil)[12:])
 }
 
-func (e *EVM) InvokeContractByFuncName(ctx context.Context, fromAddr address.Address, idAddr address.Address, funcSignature string, inputData []byte) []byte {
+func (e *EVM) InvokeContractByFuncName(ctx context.Context, fromAddr address.Address, idAddr address.Address, funcSignature string, inputData []byte) ([]byte, *api.MsgLookup, error) {
 	entryPoint := CalcFuncSignature(funcSignature)
-	wait := e.InvokeSolidity(ctx, fromAddr, idAddr, entryPoint, inputData)
-	require.True(e.t, wait.Receipt.ExitCode.IsSuccess(), "contract execution failed: %d", wait.Receipt.ExitCode)
+	wait, err := e.InvokeSolidity(ctx, fromAddr, idAddr, entryPoint, inputData)
+	if err != nil {
+		return nil, wait, err
+	}
+	if !wait.Receipt.ExitCode.IsSuccess() {
+		return nil, wait, fmt.Errorf("Contract execution failed - %v", wait.Receipt.ExitCode)
```

Comment:
Idiomatically, golang error messages are not supposed to start with a capital letter. Don't ask me why, I don't know, I'm just a messenger from a long line of go programmers